### PR TITLE
store,executor: tiny changes, consider ctx

### PR DESCRIPTION
**Pull request by tiancaiamao**
use:

```
select {
  case <-ctx.Done();
}
```

so we can cancel the computation.

### CODE REVIEW

File: store/tikv/client.go [hunk 1]
```diff
@@ -92,6 +92,12 @@ func (c *rpcClient) SendKVReq(ctx goctx.Context, addr string, req *kvrpcpb.Reque
 	start := time.Now()
 	defer func() { sendReqHistogram.WithLabelValues("kv").Observe(time.Since(start).Seconds()) }()
 
+	select {
```
Comment (shenli) [inline store/tikv/client.go]: Also need to modify SendCopReq?

File: store/tikv/client.go [hunk 2]
```diff
@@ -92,6 +98,12 @@ func (c *rpcClient) SendKVReq(ctx goctx.Context, addr string, req *kvrpcpb.Reque
...
```
Comment (zimulala) [inline store/tikv/client.go]: This code is duplicate with the code on the 60th line in `store/tikv/region_request.go`.

**Comment by tiancaiamao** [comment]
PTAL @disksing @shenli @zimulala

**Comment by zimulala** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**